bot/exts/utils/attachment_pastebin_uploader.py

```diff
@@ -0,0 +1,156 @@
+        encoding = re.search(r"charset=(\S+)", attachment.content_type).group(1)
+        file_content_bytes = await attachment.read()
+        file_content = file_content_bytes.decode(encoding)
+        return paste_service.PasteFile(content=file_content, name=attachment.filename)
+
+    async def wait_for_user_reaction(
+        self,
+        message: discord.Message,
+        user: discord.User,
+        emoji: str,
+        timeout: float = 60,
+    ) -> bool:
+        """Wait for `timeout` seconds for `user` to react to `message` with `emoji`."""
+        def wait_for_reaction(reaction: discord.Reaction, reactor: discord.User) -> bool:
+            return (
+                reaction.message.id == message.id
+                and str(reaction.emoji) == emoji
+                and reactor == user
+            )
+
+        await message.add_reaction(emoji)
+        log.trace(f"Waiting for {user.name} to react to {message.id} with {emoji}")
+
+        try:
+            await self.bot.wait_for("reaction_add", timeout=timeout, check=wait_for_reaction)
+        except TimeoutError:
+            log.trace(f"User {user.name} did not react to message {message.id} with {emoji}")
+            await message.clear_reactions()
+            return False
+
+        return True
+
+    @commands.Cog.listener()
+    async def on_message_delete(self, message: discord.Message) -> None:
+        """Allows us to know which messages with attachments have been deleted."""
+        self.pending_messages.discard(message.id)
+
+    @commands.Cog.listener()
+    async def on_message(self, message: discord.Message) -> None:
+        """Listens for messages containing attachments and offers to upload them to the pastebin."""
+        # Check if the message contains an embedded file and is not sent by a bot.
+        if message.author.bot or not any("charset" in a.content_type for a in message.attachments):
+            return
+
+        log.trace(f"Offering to upload attachments for {message.author} in {message.channel}, message {message.id}")
+        self.pending_messages.add(message.id)
+
+        # Offer to upload the attachments and wait for the user's reaction.
+        bot_reply = await message.reply(
+            f"Please react with {UPLOAD_EMOJI} to upload your file(s) to our "
+            f"[paste bin](<https://paste.pythondiscord.com/>), which is more accessible for some users."
+        )
+
+        permission_granted = await self.wait_for_user_reaction(bot_reply, message.author, UPLOAD_EMOJI, 60. * 3)
+
+        if not permission_granted:
+            log.trace(f"{message.author} didn't give permission to upload {message.id} content; aborting.")
+            await bot_reply.edit(content=f"~~{bot_reply.content}~~")
+            return
+
+        if message.id not in self.pending_messages:
+            log.trace(f"{message.author}'s message was deleted before the attachments could be uploaded; aborting.")
+            await bot_reply.delete()
```

Comment:
If the help thread gets locked, will the bot be able to delete its message?
Users (including moderators) cant delete messages from a thread that is locked.